### PR TITLE
Fix autostart, improve font config, add proper mouse support

### PR DESCRIPTION
**Pull request by keli5**
This ensures that the home directory is checked by default for the autostart file, and separates the font size from the font key in the config. Also adds window.setCursor and window.unsetCursor methods for graphical cursor management.

### CODE REVIEW

File: src/nativeMain/kotlin/util/Util.kt [hunk 1]
```diff
@@ -31,4 +33,21 @@ fun Display.sendClientMessage(type: Atom, window: Window, propagate: Boolean = f
 }
 
 fun Display.sendClientMessage(type: Atom, propagate: Boolean = false, eventMask: Long = NoEventMask, vararg data: Long) =
-    sendClientMessage(type, rootWindow, propagate, eventMask, *data)
\ No newline at end of file
+    sendClientMessage(type, rootWindow, propagate, eventMask, *data)
+
+fun Window.setCursor(dpy: Display, cur: UInt = 2.toUInt(), window: Window = this) {
```
Comment (zt64) [inline src/nativeMain/kotlin/util/Util.kt:38]: You could reduce this to 
```kt
fun Window.setCursor(dpy: Display, cur: UInt = 2u) { 
```
and replace usages of `window` with `this`

Comment (zt64) [inline src/nativeMain/kotlin/util/Util.kt:38]: You could reduce this to 
```kt
fun Window.setCursor(dpy: Display, cur: UInt = 2u) { 
```
and replace usages of `window` with `this`

File: src/nativeMain/kotlin/util/Util.kt [hunk 2]
```diff
@@ -31,4 +33,21 @@ fun Display.sendClientMessage(type: Atom, window: Window, propagate: Boolean = f
...
+
+fun Window.setCursor(dpy: Display, cur: UInt = 2.toUInt(), window: Window = this) { 
+    var cursorInitialized = windowCursorInitialized[window]
+    if (cursorInitialized != true) {
+        windowCursors.put(window, XCreateFontCursor(dpy.ptr, cur))
+        XDefineCursor(dpy.ptr, window, windowCursors[window]!!)
+        windowCursorInitialized.put(window, true)
+    }
+}
+fun Window.unsetCursor(dpy: Display, window: Window = this) {
```
Comment (zt64) [inline src/nativeMain/kotlin/util/Util.kt:46]: You could also remove `window` argument and replace usages of it with `this`. Since if you have access to an instance of a window you can use this extension method directly

File: src/nativeMain/kotlin/Prism.kt [hunk 3]
```diff
@@ -6,7 +6,7 @@ import kotlin.system.exitProcess
 
 class Prism(
     private val config: Config,
-    private val dpy: Display
+    private val dpy: Display,
```
Comment (zt64) [inline src/nativeMain/kotlin/Prism.kt:9]: Trailing comma

File: src/nativeMain/kotlin/util/Util.kt [hunk 4]
```diff
@@ -10,6 +10,8 @@ import xlib.*
 
 fun getConfigDir(): String? = getenv("XDG_CONFIG_HOME")?.toKString() ?: getenv("HOME")?.toKString()?.plus("/.config")
 fun Boolean.toInt(): Int = if (this) 1 else 0
+var windowCursors: MutableMap<Window, Cursor> = mutableMapOf()
+var windowCursorInitialized: MutableMap<Window, Boolean> = mutableMapOf()
```
Comment (zt64) [inline src/nativeMain/kotlin/util/Util.kt:14]: Are these for using later?

**Review by zt64 — CHANGES_REQUESTED** [review]

File: src/nativeMain/kotlin/util/Util.kt (hunk 4 above)
Comment (keli5) [reply src/nativeMain/kotlin/util/Util.kt:14]: windowCursorInitialized remains to stop the WM from continually trying to recreate the cursor if it's used in the wrong place, but windowCursors can be removed

File: src/nativeMain/kotlin/util/Util.kt [hunk 5]
```diff
@@ -31,4 +32,18 @@ fun Display.sendClientMessage(type: Atom, window: Window, propagate: Boolean = f
 }
 
 fun Display.sendClientMessage(type: Atom, propagate: Boolean = false, eventMask: Long = NoEventMask, vararg data: Long) =
-    sendClientMessage(type, rootWindow, propagate, eventMask, *data)
\ No newline at end of file
+    sendClientMessage(type, rootWindow, propagate, eventMask, *data)
+
+public fun Window.setCursor(dpy: Display, cur: UInt = 2.toUInt()) {
```
Comment (zt64) [inline src/nativeMain/kotlin/util/Util.kt:37]: `public` access modifier isn't necessary here, would look cleaner without.
`2.toUInt()` can be replaced with `2u`

File: src/nativeMain/kotlin/util/Util.kt [hunk 6]
```diff
@@ -31,4 +32,18 @@ fun Display.sendClientMessage(type: Atom, window: Window, propagate: Boolean = f
...
+    sendClientMessage(type, rootWindow, propagate, eventMask, *data)
+
+public fun Window.setCursor(dpy: Display, cur: UInt = 2.toUInt()) { 
+    var cursorInitialized = windowCursorInitialized[this]
+    if (cursorInitialized != true) {
+        XDefineCursor(dpy.ptr, this, XCreateFontCursor(dpy.ptr, cur))
+        windowCursorInitialized.put(this, true)
+    }
+}
+public fun Window.unsetCursor(dpy: Display) {
```
Comment (zt64) [inline src/nativeMain/kotlin/util/Util.kt:44]: `public` access modifier isn't necessary here, would look cleaner without.

**Review by zt64 — CHANGES_REQUESTED** [review]

**Review by zt64 — APPROVED** [review]
